src/VecSim/utils/vec_utils.cpp

```diff
@@ -168,3 +176,54 @@ size_t VecSimType_sizeof(VecSimType type) {
     }
     return 0;
 }
+
+static void concat_results(VecSimQueryResult *dst, VecSimQueryResult *src,
+                           const VecSimQueryResult *src_end, size_t limit) {
+    while (limit && src != src_end) {
+        array_append(dst, *src);
```

Comment:
I don't agree. It is safe because we know we allocated enough memory, and the compiler will optimize the `array_append` to not include a `mov` for each append call